lib/teleterm/services/desktop/directorysharing.go

```diff
@@ -168,22 +168,28 @@ func (d *DirectoryAccess) Read(relativePath string, offset int64, buf []byte) (i
 		return 0, trace.Wrap(err)
 	}
 
-	opened, err := os.Open(path)
+	file, err := os.Open(path)
 	if err != nil {
 		return 0, trace.Wrap(err)
 	}
-	defer opened.Close()
+	defer func() {
+		closeErr := file.Close()
+		if closeErr != nil && err == nil {
+			// Only update err if no previous error occurred.
+			err = trace.Wrap(closeErr)
```

Comment:
This won't work as expected because it needs to use named return values.

Here's [a faulty example](https://go.dev/play/p/ifdYQ6XVE5j) and [a working example](https://go.dev/play/p/i5_b-yejdP0).

Reply:
Thanks! I didn't know that named returns make a difference here.